usr/share/rear/skel/default/etc/scripts/system-setup

```diff
@@ -66,20 +66,28 @@ if rear_debug ; then
     esac
 fi
 
+# Because "rear recover" won't work without default.conf
+# we abort when there is no default.conf (or when it is empty),
+# cf. https://github.com/rear/rear/pull/3070#discussion_r1389361339
+if ! test -s /usr/share/rear/conf/default.conf ; then
+    echo -e "\nERROR: ReaR recovery cannot work without /usr/share/rear/conf/default.conf - exiting\n"
+    exit 1
+fi
+
+# Set SECRET_OUTPUT_DEV because secret default values are set via
+#   { VARIABLE='secret value' ; } 2>>/dev/$SECRET_OUTPUT_DEV
+# cf. https://github.com/rear/rear/pull/3034#issuecomment-1691609782
+SECRET_OUTPUT_DEV="null"
 # Sourcing of the conf/default.conf file as we may use some defined variables or arrays
 # E.g. UDEV_NET_MAC_RULE_FILES is used by script 55-migrate-network-devices.sh
-if test -f /usr/share/rear/conf/default.conf ; then
-    # Set SECRET_OUTPUT_DEV because secret default values are set via
-    #   { VARIABLE='secret value' ; } 2>>/dev/$SECRET_OUTPUT_DEV
-    # cf. https://github.com/rear/rear/pull/3034#issuecomment-1691609782
-    SECRET_OUTPUT_DEV="null"
-    source /usr/share/rear/conf/default.conf
-fi
+source /usr/share/rear/conf/default.conf || echo -e "\n'source /usr/share/rear/conf/default.conf' failed with exit code $?\n"
 # Sourcing user and rescue configuration as we need some variables
 # (EXCLUDE_MD5SUM_VERIFICATION right now and other variables in the system setup scripts):
 # The order of sourcing should be 'site' then 'local' and as last 'rescue'
 for conf in site local rescue ; do
-    test -f /etc/rear/$conf.conf && source /etc/rear/$conf.conf
+    if test -s /etc/rear/$conf.conf ; then
```

Comment:
**Side note**, this code maybe also should support additional configuration files specified by `-C`?

Maybe the `CONFIG_APPEND_FILES_PATHS[@]` variable that we set in `rear` can help? Although it seems that we don't carry that over into the rescue system as of now.

Reply:
Yes, I think this is bug that is similar as
https://github.com/rear/rear/issues/2942
(which I will implement when time permits)
but it is a different and separated issue
that I filed right now as
https://github.com/rear/rear/issues/3083

I do not like to mix up that separated issue
with the REBOOT_COMMANDS issue here ("KSIS").

I do already here some bits more than implementing
only what actually belongs to REBOOT_COMMANDS
but for those bits more I still feel OK because
implementing those bits "just by the way" here
looks simple enough to not cause real new issues.

In general I do not want to let an issue or a pull request
grow too much until I lost track(s) so that finally
I may have to give up the whole thing.

Reply:
What about removing the support for -C in a future (possibly incompatible) ReaR 3.0? It would simplify things. (-c should be ok given https://github.com/rear/rear/blob/6fa0df56488a67d9a63d6b3c17a8d79bd21d9772/usr/share/rear/build/GNU/Linux/100_copy_as_is.sh#L23 )

Reply:
'-C' is there for "Using Multiple Backups for Relax-and-Recover"
as described on
https://github.com/rear/rear/blob/master/doc/user-guide/11-multiple-backups.adoc
See at the bottom of that text why running multiple backups
and in particular multiple restores in parallel could
be very helpful in case of emergency and time pressure
during a real disaster recovery.

Reply:
Only as pre-information FYI:
Inspired by this issue I am thinking about a new
RECOVERY_COMMANDS array to specify the command(s)
that should be run to recover the system
in automated or unattended modes
which is by default "rear recover".

This way the user can specify the right commands
for his particular usage of '-C'.

Reply:
> '-C' is there for "Using Multiple Backups for Relax-and-Recover" as described on https://github.com/rear/rear/blob/master/doc/user-guide/11-multiple-backups.adoc See at the bottom of that text why running multiple backups and in particular multiple restores in parallel could be very helpful in case of emergency and time pressure during a real disaster recovery. This is the only currently intended (i.e. supported) use case and reason behind '-C'.

Would it simplify the code if `-C` stops supporting paths and supports only relative file names under `/etc/rear` (or the directory specified by `-c`) ? This way we would be sure that all config files are in the same directory.

I am also thinking that the directory specified by `-c` could be symlinked to `/etc/rear` instead of having its content copied, just as we do with `$SHARE_DIR` and `$VAR_DIR`.

Reply:
@pcahyna 
let me first implement
https://github.com/rear/rear/pull/3089
which intends to fix
https://github.com/rear/rear/issues/2942
at least for the primarily intended use case of '-C'.
Then I may got a bit better understanding
how to deal with '-C' in practice during recovery.

Personally I never used '-c'.
I don't know what its intended use case is.
I found '-c' only mentioned in
doc/rear-release-notes.txt
at
```
Issue Description: usage of an alternative configuration
directory is different in mkbackup or recover mode

Using rear -v -c /etc/rear/mydir mkbackup works fine in
production, but when you try (once booted from rescue image)
rear -v -c /etc/rear/mydir recover
it will fail.

-   Workaround:

The configuration files are copied to /etc/rear/
into the rescue image, so you
need to type: rear -v recover
See issue #512
```
see
https://github.com/rear/rear/issues/512#issuecomment-68051604

Accordingly it seems it is not possible
to have more than one config directory
like /etc/rear_this and /etc/rear_that
and use them as needed during recovery.
So when only one can be used during recovery
why is more than one needed during mkrescue/mkbackup?
If my understanding is right
I think the current '-c' implementation
causes false expectations.
If '-c' was not there a user who needs
/etc/rear_this and /etc/rear_that could set
the actually used one as symlink target for /etc/rear
and then it would be clear to the user
that recovery will use the config directory
that he had used during mkrescue/mkbackup.